staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/filter_test.go

```diff
@@ -183,6 +191,37 @@ func TestFilter(t *testing.T) {
 		testPerCallLimit uint64
 		namespaceObject  *corev1.Namespace
 	}{
+		{
+			name: "test escaping",
+			validations: []ExpressionAccessor{
+				&condition{
+					Expression: "object.metadata.labels.helm__dot__sh__slash__chart == 'other'",
```

Comment:
Let's also test that the string is only escaped when used as a val, e.g. `object.metadata.labels['helm.sh/chart'] == 'other'.

Reply:
Ha that is a good catch! I realize that we don't support [] selection in CRD.. This one seems tricky hence we didn't have a easy way to differentiate them. The possible way forward I could think of is to reconstruct a new ast with properly escaped map key, however, it would require the ast struct exported. https://github.com/google/cel-go/blob/fc501a3a9c46dca28c79477a0fd8ae48a2df8ec8/common/ast/ast.go#L24-L30
 Not sure if that is something cel would consider? @TristonianJones